relational_operators/DeleteOperator.cpp

```diff
@@ -81,19 +85,25 @@ void DeleteWorkOrder::execute(QueryContext *query_context,
 
   // TODO(harshad): Stream the block ID only if the predicate returned at least
   // one match in the StorageBlock.
-  ForemanMessage message(ForemanMessage::DataPipelineMessage(delete_operator_index_,
-                                                             input_block_id_,
-                                                             rel_id_));
+  serialization::DataPipelineMessage proto;
+  proto.set_operator_index(delete_operator_index_);
+  proto.set_block_id(input_block_id_);
+  proto.set_relation_id(rel_id_);
+
+  const std::size_t proto_length = proto.ByteSize();
+  char *proto_bytes = static_cast<char*>(std::malloc(proto_length));
```

Comment:
I agree with @zuyu's explanation.  It might be better if we add this explanation as a comment (in one of the several places where the pattern is used), so that the rationale for using heap memory is clear to the reader.